### PR TITLE
Add compression card to pool dashboard

### PR DESCRIPTION
**Pull request by aruniiird**
Adding compression card to pool dashboard, that means this PR is on top of @GowthamShanmugam 's PR#9423.

### CODE REVIEW

**Comment by aruniiird** [comment]
@GowthamShanmugam , @afreen23 , @bipuladh , @cloudbehl , please take a look.

File: frontend/packages/ceph-storage-plugin/src/queries/ceph-queries.ts [hunk 1]
```diff
@@ -233,6 +237,15 @@ export const getPoolQuery = (poolNames: string[], queryName: string) => {
     [StorageDashboardQuery.POOL_UTILIZATION_THROUGHPUT_QUERY]: `(rate(ceph_pool_wr_bytes[1m]) + rate(ceph_pool_rd_bytes[1m])) * on (pool_id) group_left(name)ceph_pool_metadata{name=~'${poolNames.join(
       '|',
     )}'}`,
+    [StorageDashboardQuery.POOL_COMPRESSION_SAVINGS]: `(ceph_pool_compress_under_bytes - ceph_pool_compress_bytes_used) * on (pool_id) group_left(name)ceph_pool_metadata{name=~'${poolNames.join(
+      '|',
+    )}'}`,
+    [StorageDashboardQuery.POOL_COMPRESSION_ELIGIBILITY]: `((sum(ceph_pool_compress_under_bytes > 0) / sum(ceph_pool_stored_raw and ceph_pool_compress_under_bytes > 0)) * 100) * on (pool_id) group_left(name)ceph_pool_metadata{name=~'${poolNames.join(
```
Comment (GowthamShanmugam) [inline frontend/packages/ceph-storage-plugin/src/queries/ceph-queries.ts:243]: Do we really need SUM here, it is a single pool query so I doubting some is not makesense. May be i am wrong here, you better check with @anmolsachan once

File: frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      getPoolQuery([name], StorageDashboardQuery.POOL_COMPRESSION_ELIGIBILITY),
+      getPoolQuery([name], StorageDashboardQuery.POOL_COMPRESSION_RATIO),
+    ],
+    [name],
+  );
+
+  const [values, loading] = usePrometheusQueries(queries, parser as any);
+  const poolCompressionSavings = (values?.[0] ?? 0) as number;
+  const poolCompressionEligibility = (values?.[1] ?? 0) as number;
+  const poolCompressionRatio = (values?.[2] ?? 0) as number;
```
Comment (GowthamShanmugam) [inline frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx:40]: None of the metrics are humanized here
you have to use 
humanizePercentage  for poolCompressionEligibility
humanizeBinaryBytes for  poolCompressionSavings
is poolCompressionRatio is just a number? (no need to do like a:b)?

File: frontend/packages/ceph-storage-plugin/src/queries/ceph-queries.ts [hunk 3]
```diff
@@ -233,6 +237,15 @@ export const getPoolQuery = (poolNames: string[], queryName: string) => {
...
+      '|',
+    )}'}`,
+    [StorageDashboardQuery.POOL_COMPRESSION_RATIO]: `((ceph_pool_compress_under_bytes / ceph_pool_compress_bytes_used > 0) and on(pool_id) (((ceph_pool_compress_under_bytes > 0) / ceph_pool_stored_raw) * 100 > 0.5)) * on (pool_id) group_left(name)ceph_pool_metadata{name=~'${poolNames.join(
+      '|',
```
Comment (GowthamShanmugam) [inline frontend/packages/ceph-storage-plugin/src/queries/ceph-queries.ts:247]: I am wondering these queries are correct, Please check with @anmolsachan  about these queries

**Comment by GowthamShanmugam** [comment]
Please resolve the conflicts  and add some screenshots also

File: frontend/packages/ceph-storage-plugin/src/queries/ceph-queries.ts (hunk 1 above)
Comment (aruniiird) [reply frontend/packages/ceph-storage-plugin/src/queries/ceph-queries.ts]: Corrected

File: frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+  return (
+    <DashboardCard data-test-id="compression-details-card">
+      <DashboardCardHeader>
+        <DashboardCardTitle>{t('ceph-storage-plugin~Compression')}</DashboardCardTitle>
+      </DashboardCardHeader>
+      <DashboardCardBody>
+        <DetailsBody>
+          <DetailItem isLoading={!obj} title={t('ceph-storage-plugin~Compression Status')}>
+            {!compressionEnabled ? 'Disabled' : obj.spec.compressionMode}
```
[the remark this answers is not in the document]
Comment (aruniiird) [reply frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx]: Corrected.

File: frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+          </DetailItem>
+          {compressionEnabled && (
+            <>
+              <DetailItem
```
[the remark this answers is not in the document]
Comment (aruniiird) [reply frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx:62]: Added the title.

File: frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx (hunk 2 above)
Comment (aruniiird) [reply frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx:45]: Humanized all the numbers. Yes, Ratio is just a number and not something shown in `x : y` format.

File: frontend/packages/ceph-storage-plugin/src/queries/ceph-queries.ts (hunk 3 above)
Comment (aruniiird) [reply frontend/packages/ceph-storage-plugin/src/queries/ceph-queries.ts:247]: Checked with @anmolsachan . Queries generally look good. But as per @anmolsachan , some general changes (which is general for all the querries, not just for compression querries), we may have to add prometheus recording rules, but on a later stage.

Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/queries/ceph-queries.ts:247]: ack

File: frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+import { BlockPoolDashboardContext } from './block-pool-dashboard-context';
+import { getPoolQuery, StorageDashboardQuery } from '../../../queries/ceph-queries';
+
+const parser = compose((val) => val?.[0]?.y, getInstantVectorStats);
+
+export const CompressionDetailsCard: React.FC = () => {
+  const { t } = useTranslation();
+  const { obj } = React.useContext(BlockPoolDashboardContext);
+  const { compressionMode } = obj.spec;
```
[the remark this answers is not in the document]
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx:28]: It is better you use: const compressionMode  = obj.spec?.compressionMode; it will fail if the pool is created from CLI and if the user immediately opens the pool dashboard page after pool creation.

File: frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx (hunk 2 above)
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx:45]: ack

File: frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+  const poolCompressionRatio = (values?.[2] ?? 0) as number;
+
+  return (
+    <DashboardCard data-test-id="compression-details-card">
+      <DashboardCardHeader>
+        <DashboardCardTitle>{t('ceph-storage-plugin~Compression')}</DashboardCardTitle>
+      </DashboardCardHeader>
+      <DashboardCardBody>
+        <DetailsBody>
+          <DetailItem isLoading={!obj} title={t('ceph-storage-plugin~Compression Status')}>
```
Comment (GowthamShanmugam) [inline frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx:54]: <DetailItem isLoading={!obj} title={t('ceph-storage-plugin~Compression Status')}>

cahnge to 

<DetailItem isLoading={!obj} title={t('ceph-storage-plugin~Compression status')}>

File: frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+            {!compressionEnabled ? 'Disabled' : 'Enabled'}
+          </DetailItem>
+          {compressionEnabled && (
+            <>
+              <DetailItem isLoading={loading} title={t('ceph-storage-plugin~Storage Efficiency')}>
```
Comment (GowthamShanmugam) [inline frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx:59]: <DetailItem isLoading={loading} title={t('ceph-storage-plugin~Storage Efficiency')}>

it should be 

<DetailItem isLoading={loading} title={t('ceph-storage-plugin~Storage efficiency')}>

Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx:59]: <DetailItem isLoading={loading} title={t('ceph-storage-plugin~Storage Efficiency')} />

File: frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+                {}
+              </DetailItem>
+              <DetailItem
+                isLoading={loading}
+                title={t('ceph-storage-plugin~Compression Eligibility')}
```
Comment (GowthamShanmugam) [inline frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx:64]: title={t('ceph-storage-plugin~Compression eligibility')}

File: frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+              >
+                {humanizePercentage(poolCompressionEligibility).string}
+              </DetailItem>
+              <DetailItem isLoading={loading} title={t('ceph-storage-plugin~Compression Ratio')}>
```
Comment (GowthamShanmugam) [inline frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx:68]: title={t('ceph-storage-plugin~Compression ratio')

File: frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+                {humanizeNumber(poolCompressionRatio)}
+              </DetailItem>
+              <DetailItem isLoading={loading} title={t('ceph-storage-plugin~Compression Savings')}>
+                {humanizeBinaryBytes(poolCompressionSavings).string}
+              </DetailItem>
```
Comment (GowthamShanmugam) [inline frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx:73]: title={t('ceph-storage-plugin~Compression savings')

File: frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (aruniiird) [reply frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx]: Done.

File: frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx (hunk 7 above)
Comment (aruniiird) [reply frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx]: Changed...

File: frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx (hunk 8 above)
Comment (aruniiird) [reply frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx]: Changed

File: frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx (hunk 9 above)
Comment (aruniiird) [reply frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx]: Changed

File: frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx (hunk 10 above)
Comment (aruniiird) [reply frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx]: Changed

File: frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx (hunk 11 above)
Comment (aruniiird) [reply frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx:73]: Changed

**Comment by GowthamShanmugam** [comment]
Please share some screenshot here

File: frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx (hunk 2 above)
Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx:45]: Rather than defaulting to 0. We should be showing :
1. "not available" : when no values/ empty values
2.  when its actually 0

File: frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx [hunk 12]
```diff
@@ -0,0 +1,80 @@
...
+      </DashboardCardHeader>
+      <DashboardCardBody>
+        <DetailsBody>
+          <DetailItem isLoading={!obj} title={t('ceph-storage-plugin~Compression status')}>
+            {!compressionEnabled ? 'Disabled' : 'Enabled'}
+          </DetailItem>
+          {compressionEnabled && (
+            <>
+              <DetailItem isLoading={loading} title={t('ceph-storage-plugin~Storage efficiency')}>
+                {''}
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx:60]: why empty ?
can you share screenshot as well ?

**Review by afreen23 — CHANGES_REQUESTED** [review]

File: frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx (hunk 12 above)
Comment (aruniiird) [reply frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx]: The compression card itself

Screenshot from 2021-07-21 13-20-35

Compression card in the dashboard page

Screenshot from 2021-07-21 13-19-40

Comment (aruniiird) [reply frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx]: @GowthamShanmugam , @afreen23 , please take a look at the screen shot

Comment (aruniiird) [reply frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx]: @afreen23 , for your above comment (https://github.com/openshift/console/pull/9507#discussion_r673313723), I have updated the code a bit differently.

Removed the default ZERO value, as we won't enter into the default state if the query is still being loaded (or we hit any query error). Please take a look.

File: frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx (hunk 2 above)
Comment (aruniiird) [reply frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx]: @afreen23 , for your above comment (https://github.com/openshift/console/pull/9507#discussion_r673313723), I have updated the code a bit differently.

Removed the default ZERO value, as we won't enter into the default state if the query is still being loaded (or we hit any query error). Please take a look.

File: frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+    error: loadError || !!poolCompressionSavings,
+    title: t('ceph-storage-plugin~Compression savings'),
+    getStats: () => humanizeBinaryBytes(poolCompressionSavings).string,
+    infoText: 'shows the total savings gained from compression, including the replicas',
+  };
+
+  const compressionRatioProps = {
+    stats: Number(poolCompressionRatio),
+    isLoading: loading,
+    error: loadError || !!poolCompressionRatio,
```
Comment (GowthamShanmugam) [inline frontend/packages/ceph-storage-plugin/src/components/block-pool/dashbaord/compression-details-card.tsx:70]: @afreen23 Internally error will takecare of putting "Not available" on screen.

**Review by vbnrh — CHANGES_REQUESTED** [review]
@aruniiird In the compression card, the spacing between title and the contents are not uniform . Please check

**Comment by aruniiird** [comment]
> @aruniiird In the compression card, the spacing between title and the contents are not uniform . Also the text size for the contents seem a bit bigger than the titles itself. I wonder if you could make the title headings slightly bigger. Please check

Updated the code (thanks to @GowthamShanmugam ), here are the screenshots,

Screenshot from 2021-07-23 18-51-54

Screenshot from 2021-07-23 18-52-22

**Comment by vbnrh** [comment]
Looks better than the last one but there are still couple of issues 
- There is a larger gap between Storage efficiency and compression eligibility. 
If we check above there is not that much difference in the Compression status and enabled. They both should be uniform 
- The heading should be larger than the contents. `Storage efficiency` and `Compression status` should have larger text size but here it looks the same and even smaller than its body.
- There is a large amount of space after 2nd columns in storage efficiency. I would suggest to add larger margin to move it at the end of the card.

File: frontend/packages/ceph-storage-plugin/locales/en/ceph-storage-plugin.json [hunk 14]
```diff
@@ -32,6 +32,13 @@
...
+  "Compression ratio": "Compression ratio",
+  "Compression savings": "Compression savings",
+  "<0><0><0> What does each state mean?</0></0><1><0>Starting replay:</0> Initiating image replication process.</1><2><0>Replaying:</0> Image replication is ongoing or idle between clusters.</2><3><0>Stopping replay:</0> Image replication process is shutting down.</3><4><0>Stopped:</0> Image replication process has shut down.</4><5><0>Error:</0> Image replication process stopped due to an error. </5><6><0>Unknown:</0> Unable to determine image state due to an error. Check your network connection and remote cluster RBD mirror daemon.</6></0>": "<0><0><0> What does each state mean?</0></0><1><0>Starting replay:</0> Initiating image replication process.</1><2><0>Replaying:</0> Image replication is ongoing or idle between clusters.</2><3><0>Stopping replay:</0> Image replication process is shutting down.</3><4><0>Stopped:</0> Image replication process has shut down.</4><5><0>Error:</0> Image replication process stopped due to an error. </5><6><0>Unknown:</0> Unable to determine image state due to an error. Check your network connection and remote cluster RBD mirror daemon.</6></0>",
```
Comment (GowthamShanmugam) [inline frontend/packages/ceph-storage-plugin/locales/en/ceph-storage-plugin.json:40]: Why this change is in your commit? you better rebase once and run i18n

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/block-pool/compression-details-card.tsx [hunk 15]
```diff
@@ -0,0 +1,102 @@
...
+      <DashboardCardHeader>
+        <DashboardCardTitle>{t('ceph-storage-plugin~Compression')}</DashboardCardTitle>
+      </DashboardCardHeader>
+      <DashboardCardBody>
+        <DetailsBody>
+          <DetailItem isLoading={!obj} title={t('ceph-storage-plugin~Compression status')}>
+            {!compressionEnabled ? 'Disabled' : 'Enabled'}
+          </DetailItem>
+        </DetailsBody>
+        <DetailsBody>
```
Comment (GowthamShanmugam) [inline frontend/packages/ceph-storage-plugin/src/components/dashboards/block-pool/compression-details-card.tsx:88]: remove \<DetailsBody\> from here

**Review by afreen23 — APPROVED** [review]

**Comment by aruniiird** [comment]
> Looks better than the last one but there are still couple of issues
> 
>     * There is a larger gap between Storage efficiency and compression eligibility.

Because in the mock-up there is a larger gap. Please see the side by side image below.

>       If we check above there is not that much difference in the Compression status and enabled. They both should be uniform
> 
>     * The heading should be larger than the contents. `Storage efficiency` and `Compression status` should have larger text size but here it looks the same and even smaller than its body.

Please see the mock-up, heading/title is not that bigger than the value. Heading is just bold.

> 
>     * There is a large amount of space after 2nd columns in storage efficiency. I would suggest to add larger margin to move it at the end of the card.

True, not an expert here, will try to push the value part to the end of the card.


............. **Mockup Card** ...................................................................... **Actual Card** ..........

Screenshot from 2021-07-23 23-59-11

File: frontend/packages/ceph-storage-plugin/locales/en/ceph-storage-plugin.json (hunk 14 above)
Comment (aruniiird) [reply frontend/packages/ceph-storage-plugin/locales/en/ceph-storage-plugin.json]: Removed and ran `yarn i18n`. Rebased on top of latest PR#9423.

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/block-pool/compression-details-card.tsx (hunk 15 above)
Comment (aruniiird) [reply frontend/packages/ceph-storage-plugin/src/components/dashboards/block-pool/compression-details-card.tsx:88]: Made some changes according to @vbnrh comments. Please take a look (and please see the screenshots provided below as well)

**Comment by aruniiird** [comment]
@vbnrh , I have updated the code and here are the screen shots,

........... **Mockup Card** ......................................................... **Actual Card** ..................

Screenshot from 2021-07-24 01-15-28


Over all picture

Screenshot from 2021-07-24 01-00-30

**Comment by aruniiird** [comment]
Addressed all the conflicts and rebased to the latest upstream master.
@afreen23 , @vbnrh , @GowthamShanmugam , please take a look...

**Review by GowthamShanmugam — APPROVED** [review]

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/block-pool/compression-details-card.tsx [hunk 16]
```diff
@@ -0,0 +1,104 @@
...
+
+  return (
+    <DashboardCard data-test-id="compression-details-card">
+      <DashboardCardHeader>
+        <DashboardCardTitle>{t('ceph-storage-plugin~Compression')}</DashboardCardTitle>
+      </DashboardCardHeader>
+      <DashboardCardBody>
+        <DetailsBody>
+          <DetailItem isLoading={!obj} title={t('ceph-storage-plugin~Compression status')}>
+            {!compressionEnabled ? 'Disabled' : 'Enabled'}
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/dashboards/block-pool/compression-details-card.tsx:85]: externalize this
```suggestion
            {!compressionEnabled ? t('Disabled') : t('Enabled')}
```

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/block-pool/compression-details-card.tsx [hunk 17]
```diff
@@ -0,0 +1,104 @@
...
+  const poolCompressionEligibility = values?.[1];
+  const poolCompressionRatio = values?.[2];
+
+  const compressionEligibilityProps = {
+    stats: Number(poolCompressionEligibility),
+    isLoading: loading,
+    error: loadError || !poolCompressionEligibility,
+    title: t('ceph-storage-plugin~Compression eligibility'),
+    getStats: () => humanizePercentage(poolCompressionEligibility).string,
+    infoText: 'indicates the percentage of incoming data that is compressible',
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/dashboards/block-pool/compression-details-card.tsx:55]:
```suggestion
    infoText: t('Compression eligibility indicates the percentage of incoming data that is compressible',)
```

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/block-pool/compression-details-card.tsx [hunk 18]
```diff
@@ -0,0 +1,104 @@
...
+  };
+
+  const compressionSavingsProps = {
+    stats: Number(poolCompressionSavings),
+    isLoading: loading,
+    error: loadError || !poolCompressionSavings,
+    title: t('ceph-storage-plugin~Compression savings'),
+    getStats: () => humanizeBinaryBytes(poolCompressionSavings).string,
+    infoText:
+      'indicates the total savings gained from compression for this pool, including replicas',
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/dashboards/block-pool/compression-details-card.tsx:65]:
```suggestion
      'Compression savings indicates the total savings gained from compression for this pool, including replicas',
```

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/block-pool/compression-details-card.tsx [hunk 19]
```diff
@@ -0,0 +1,104 @@
...
+  };
+
+  const compressionRatioProps = {
+    stats: Number(poolCompressionRatio),
+    isLoading: loading,
+    error: loadError || !poolCompressionRatio,
+    title: t('ceph-storage-plugin~Compression ratio'),
+    getStats: () => `${humanizeNumber(poolCompressionRatio).string}:1`,
+    infoText: 'indicates the achieved compression on eligible data for this pool',
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/dashboards/block-pool/compression-details-card.tsx:74]:
```suggestion
    infoText: 'Compression ratio indicates the achieved compression on eligible data for this pool',
```

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/block-pool/compression-details-card.tsx [hunk 20]
```diff
@@ -0,0 +1,104 @@
...
+import { getPoolQuery, StorageDashboardQuery } from '../../../queries/ceph-queries';
+
+const parser = compose((val) => val?.[0]?.y, getInstantVectorStats);
+
+export const CompressionDetailsCard: React.FC = () => {
+  const { t } = useTranslation();
+  const { obj } = React.useContext(BlockPoolDashboardContext);
+  const compressionMode = obj.spec?.compressionMode;
+  const compressionEnabled = compressionMode !== 'none';
+  const { name } = obj.metadata;
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/dashboards/block-pool/compression-details-card.tsx:32]:
```suggestion
  const { obj } = React.useContext(BlockPoolDashboardContext);
  
  const compressionMode = obj.spec?.compressionMode;
  const compressionEnabled = compressionMode !== 'none';
  const { name } = obj.metadata;
```

**Review by afreen23 — CHANGES_REQUESTED** [review]

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/block-pool/compression-details-card.tsx (hunk 20 above)
Comment (aruniiird) [reply frontend/packages/ceph-storage-plugin/src/components/dashboards/block-pool/compression-details-card.tsx]: Done

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/block-pool/compression-details-card.tsx (hunk 19 above)
Comment (aruniiird) [reply frontend/packages/ceph-storage-plugin/src/components/dashboards/block-pool/compression-details-card.tsx]: Done

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/block-pool/compression-details-card.tsx (hunk 18 above)
Comment (aruniiird) [reply frontend/packages/ceph-storage-plugin/src/components/dashboards/block-pool/compression-details-card.tsx]: Done

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/block-pool/compression-details-card.tsx (hunk 17 above)
Comment (aruniiird) [reply frontend/packages/ceph-storage-plugin/src/components/dashboards/block-pool/compression-details-card.tsx]: Done

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/block-pool/compression-details-card.tsx (hunk 16 above)
Comment (aruniiird) [reply frontend/packages/ceph-storage-plugin/src/components/dashboards/block-pool/compression-details-card.tsx]: Done

**Comment by aruniiird** [comment]
@afreen23 , addressed all the review comments. Please take a look...

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/block-pool/compression-details-card.tsx (hunk 15 above)
Comment (bipuladh) [reply frontend/packages/ceph-storage-plugin/src/components/dashboards/block-pool/compression-details-card.tsx:96]:
```suggestion
```

**Review by bipuladh — CHANGES_REQUESTED** [review]

**Review by afreen23 — COMMENTED** [review]
/lgtm

Please resolve https://github.com/openshift/console/pull/9507#discussion_r676096052 in a followup PR. Functionality looks good hence tagging.

**Comment by aruniiird** [comment]
@bipuladh , the suggestion provided here, https://github.com/openshift/console/pull/9507#discussion_r676096052, if added will change the way the card look.

........**Your suggestion** ..........................................................................  **Current**...................... 
image

Tag <DetailsBody> was added intentionally. In order to avoid some confusion, I moved the `<DetailsBody>` into `compressionEnabled` part. Let me know if you want to really remove the tag.

File: frontend/packages/ceph-storage-plugin/src/components/dashboards/block-pool/compression-details-card.tsx [hunk 21]
```diff
@@ -0,0 +1,112 @@
...
+            <div>
+              <DetailsBody>
+                <DetailItem isLoading={loading} title={t('ceph-storage-plugin~Storage efficiency')}>
+                  <EfficiencyItemBody {...compressionEligibilityProps} />
+                  <EfficiencyItemBody {...compressionRatioProps} />
+                  <EfficiencyItemBody {...compressionSavingsProps} />
+                </DetailItem>
+              </DetailsBody>
+            </div>
+          </DetailsBody>
```
Comment (aruniiird) [inline frontend/packages/ceph-storage-plugin/src/components/dashboards/block-pool/compression-details-card.tsx:107]: @bipuladh , moved the `<DetailsBody>` into `compressionEnabled` true part.
